### ISSUE TITLE
Doesn't support HiDPI displays

### ISSUE THREAD

**Issue by evanwalsh** [angel2d/angel2d#1]
Hello! I was testing out Angel and I noticed the demos all look a bit pixelated on my retina MBP. Here's a screenshot (make sure you look at the full size):

![Screen Shot 2013-04-09 at 22 09 20 ](https://f.cloud.github.com/assets/6129/360181/b58edbea-a183-11e2-8660-fc05f6ceee7f.png)

**Comment by sjml** [comment]
Thanks for the heads up. I don't have a retina display myself, so I won't be able to directly verify a fix. Would to be be able/willing to try out some solutions to to address this? 

I'm guessing we could do something similar to what's already done for retina iOS devices; just have to hook it into the Mac build.

**Comment by evanwalsh** [comment]
I'm definitely down to help however I can. Just let me know.

**Comment by evanwalsh** [comment]
It looks like there's already [some discussion on the GLFW side of things](https://github.com/elmindreda/glfw/pull/36) about supporting HiDPI displays. I also found [this fun set of changes](http://pastebin.com/6ms7tXUf) to GLFW that supposedly enables retina support, but I can't get it working on my end. Admittedly, I know very little about GLFW.

I'm interested in your opinion of what you think is best. I'd hate to see the project run a non-standard GLFW to support this, but I'd also hate for it to be held back because of third-party problems. 

I looked at how Angel handles iOS retina displays and it's not as easy as a straight port. You'd know better than anyone: is there an obvious way to enable HiDPI support in Angel without having to change any code inside of the bundled GLFW?

**Comment by sjml** [comment]
I doubt there's any way we could do it without modifying GFLW -- pretty much all of the windowing setup is done in their engine code, and I don't know if there's any good way to trick it, resolution-wise. 

Yeah, I didn't expect the iOS method to be portable, because, well, it doesn't use GLFW. :) 

I'll talk to Camilla (who maintains GLFW) about retina support; I imagine it's something on the project's radar, at the very minimum.

**Comment by sjml** [comment]
OK, sounds like retina support will be going into GLFW3, which is... kind of imminent? The timing is unclear. I'd prefer to wait on an official release of the library before including it, but if it doesn't come out soon, I'm ok with just patching the support in in the meanwhile.

**Comment by evanwalsh** [comment]
Ok. This isn't a showstopping issue for me, but I'd love to see it sooner rather than later. I'll keep tabs on this as much as I can and we can revisit it down the road.

**Comment by sjml** [comment]
I'd like to see it, too, believe me. 

And hey, if the Angel community wants to kick in and buy me a Retina MBP, I promise you it would bother me enough to have it fixed within 48 hours. :)

**Comment by elmindreda** [comment]
Support for Retina displays has been merged into GLFW 3.0 master.

**Comment by sjml** [comment]
Fantastic! Thanks for reporting in. I'll try to find some time to move
Angel over to GLFW 3 over the next week or so.

**Comment by sjml** [comment]
Evan, I've got some preliminary work on moving to GLFW3 done. Are you able to check it out an ensure it works as expected with your Retina MBP? You won't have access to the console for now... the whole thing is kind of incomplete, but I'm working on it. 

If you have some time, check out the Hg repository at GoogleCode: http://code.google.com/p/angel-engine/source/checkout

Switch over to the glfw3.0 branch, and build the Mac Xcode project. 

Let me know how it works out!

**Comment by sjml** [comment]
This now works in mainline Angel, and we know how to set it up for Angel4, so I'm gonna just close this down for now.
